export/export_pipeline/compute_collision_representation.py

```diff
@@ -0,0 +1,427 @@
+        decomposed = True
+        os.remove(mesh_path)
+    else:
+        os.makedirs(decompose_dir, exist_ok=True)
+        mesh_path = shutil.move(mesh_path, decompose_dir)
+        obj_name = os.path.splitext(mesh_filename)[0]
+        
+        # load mesh and check that it is valid
+        mesh: trimesh.Trimesh = trimesh.load(mesh_path, force='mesh')
+        if mesh.is_empty:
+            volume = 0.0
+        else:
+            try:
+                volume = mesh.bounding_box_oriented.volume
+            except ValueError:
+                volume = 0.0
+        if volume < 1e-6:
+            acd_args['decompose_method'] = 'skip'
+            print(f'{obj_name} is too small, volume = {volume:.5f}, will skip')
+        elif volume < acd_args['min_volume']:
+            acd_args['decompose_method'] = 'none'
+            print(f'{obj_name} is small, volume = {volume:.5f}, will not decompose')
+
+        # save convex hull
+        hull_mesh: trimesh.Trimesh = trimesh.convex.convex_hull(mesh)
+        hull_filename = osp.join(mesh_dir, 'convex_hull.stl')
+        hull_mesh.export(hull_filename)
+        print(f'Convex hull {hull_filename} saved.')
+        
+        if acd_args['decompose_method'] == 'coacd':
+            # Call CoACD.
+            if coacd_args['auto_threshold']:
+                model = coacd_args['threshold_model']
+                threshold = max(0.02, volume*model[0] + model[1])
+            else:
+                threshold = coacd_args['threshold']
+            mesh = coacd.Mesh(mesh.vertices, mesh.faces)
+            result = coacd.run_coacd(
+                mesh,
+                threshold=threshold,
+                max_convex_hull=coacd_args['max_convex_hull'],
+                preprocess_mode=coacd_args['preprocess_mode'],
+                preprocess_resolution=coacd_args['preprocess_resolution'],
+                resolution=coacd_args['resolution'],
+                mcts_nodes=coacd_args['mcts_nodes'],
+                mcts_iterations=coacd_args['mcts_iterations'],
+                mcts_max_depth=coacd_args['mcts_max_depth'],
+                pca=coacd_args['pca'],
+                merge=coacd_args['merge'],
+                seed=coacd_args['seed'],
+                )
+
+            mesh_parts = []
+            for vs, fs in result:
+                # remove faces with duplicate vertices
+                fs = [f for f in fs if len(set(f)) == 3]
+                # remove duplicate faces
+                fs_set = set()
+                final_fs = []
+                for f in fs:
+                    ff = frozenset(f)  # frozenset() also sorts
+                    if ff in fs_set:
+                        fs_set.remove(ff)
+                    else:
+                        fs_set.add(ff)
+                        final_fs.append(f)
+                mesh = trimesh.Trimesh(vs, final_fs)
+                if mesh.is_volume:
+                    mesh_parts.append(mesh)
+
+            object_assembled = trimesh.Scene()
```

Comment:
We call it an `object` here and a `mesh` earlier, but we're referring to the same underlying entity. I think it would be better if we made this more consistent.